actionview/lib/action_view/renderer/template_renderer.rb

```diff
@@ -41,7 +41,12 @@ def determine_template(options)
           end
           Template::Inline.new(options[:inline], "inline template", handler, locals: keys, format: format)
         elsif options.key?(:renderable)
-          Template::Renderable.new(options[:renderable])
+          renderable = options[:renderable]
+          unless renderable.respond_to?(:render_in)
```

Comment:
This still adds overhead in the happy path

Reply:
I've changed this to rescue from `NoMethodError`. Is that what you had in mind?